### PR TITLE
fixes compiler warnings FREEBIE

### PR DESCRIPTION
**Pull request by dermaaarkus**
### First time contributor checklist

- [x] I have read the [README](https://github.com/WhisperSystems/Signal-iOS/blob/master/README.md) and [CONTRIBUTING](https://github.com/WhisperSystems/Signal-iOS/blob/master/CONTRIBUTING.md) documents
- [x] I have signed the [Contributor Licence Agreement](https://whispersystems.org/cla/)

### Contributor checklist

- [x] I'm following the [code, UI and style conventions](https://github.com/WhisperSystems/Signal-iOS/blob/master/CONTRIBUTING.md#code-conventions)
- [x] My commits are rebased on the latest master branch
- [x] My commits are in nice logical chunks
- [x] My contribution is fully baked and is ready to be merged as is
- [x] I have tested my contribution on these devices:
 * Simulator iPhone 7, iOS iOS 10.3
- [x] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in my commit message

- - - - - - - - - -

### Description
I just checkout the project and fixes some compiler errors. I fixed some integer conversion warnings with explicit casting.

### CODE REVIEW

File: Signal/src/call/OWSWebRTCDataProtos.pb.m [hunk 1]
```diff
@@ -57,7 +57,7 @@ - (BOOL) isInitialized {
 }
 - (void) writeToCodedOutputStream:(PBCodedOutputStream*) output {
   if (self.hasId) {
-    [output writeUInt64:1 value:self.id];
```
Comment (michaelkirk) [inline Signal/src/call/OWSWebRTCDataProtos.pb.m]: This is a generated file. Please revert these changes.

Comment (dermaaarkus) [inline Signal/src/call/OWSWebRTCDataProtos.pb.m]: Those file changes are reverted, sorry mate! 
If there is any other possibility to fix/remove the project warnings I would appreciate it to help, because new warnings will be seen and fixed faster 😃

File: Signal/src/ViewControllers/DebugUI/DebugUIContacts.m [hunk 2]
```diff
@@ -1107,7 +1107,9 @@ + (NSString *)randomLastName
         @"Noel",
         @"Vang",
     ];
-    return values[arc4random_uniform(values.count)];
+    
+    uint32_t index = arc4random_uniform((uint32_t)values.count);
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/DebugUI/DebugUIContacts.m]: This actually still gives me a compiler warning:
 
screen shot 2017-07-08 at 5 27 07 pm

Comment (michaelkirk) [inline Signal/src/ViewControllers/DebugUI/DebugUIContacts.m]: Sorry, I was looking at the wrong line. Looks good!

**Comment by michaelkirk** [comment]
In 422336db32852a410da39c6143ecdf57ce9a397d.

Thanks!
